src/transformers/models/esm/modeling_esm.py

```
@@ -0,0 +1,1408 @@
+    add_code_sample_docstrings,
+    add_start_docstrings,
+    add_start_docstrings_to_model_forward,
+    replace_return_docstrings,
+)
+from ...modeling_outputs import (
+    BaseModelOutputWithPastAndCrossAttentions,
+    BaseModelOutputWithPoolingAndCrossAttentions,
+    CausalLMOutputWithCrossAttentions,
+    MaskedLMOutput,
+    SequenceClassifierOutput,
+    TokenClassifierOutput,
+)
+from ...modeling_utils import (
+    PreTrainedModel,
+    apply_chunking_to_forward,
+    find_pruneable_heads_and_indices,
+    prune_linear_layer,
+)
+from ...utils import logging
+from .configuration_esm import ESMConfig
+
+
+logger = logging.get_logger(__name__)
+
+_CHECKPOINT_FOR_DOC = "facebook/esm-1b"
+_CONFIG_FOR_DOC = "ESMConfig"
+_TOKENIZER_FOR_DOC = "ESMTokenizer"
+
+ESM_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "facebook/esm-1b",
+    # See all ESM models at https://huggingface.co/models?filter=esm
+]
+
+
+def rotate_half(x):
+    x1, x2 = x.chunk(2, dim=-1)
+    return torch.cat((-x2, x1), dim=-1)
+
+
+def apply_rotary_pos_emb(x, cos, sin):
+    cos = cos[:, :, : x.shape[-2], :]
+    sin = sin[:, :, : x.shape[-2], :]
+
+    return (x * cos) + (rotate_half(x) * sin)
+
+
+class RotaryEmbedding(torch.nn.Module):
+    """
+    The rotary position embeddings from RoFormer_ (Su et. al). A crucial insight from the method is that the query and
+    keys are transformed by rotation matrices which depend on the relative positions. Other implementations are
+    available in the Rotary Transformer repo_ and in GPT-NeoX_, GPT-NeoX was an inspiration .. _RoFormer:
+    https://arxiv.org/abs/2104.09864 .. _repo: https://github.com/ZhuiyiTechnology/roformer .. _GPT-NeoX:
+    https://github.com/EleutherAI/gpt-neox .. warning: Please note that this embedding is not registered on purpose, as
+    it is transformative
+        (it does not create the embedding dimension) and will likely be picked up (imported) on a ad-hoc basis
+    """
+
+    def __init__(self, dim: int, *_, **__):
```

Do we really need the not-so-pretty `*_` and `**_`? Also it means it will silently gobble anything without telling the user there is a typo?

---

Removed!